Rating/mec_bpc.py

```diff
@@ -0,0 +1,109 @@
+import json
+from time import sleep
+from tqdm import tqdm
+import openai
+from string import Template
+from typing import Optional, List, Union
+
+
+MAX_API_RETRY = 5
+REQ_TIME_GAP = 5
+
+
+def clean_continuation(rating):
+    rating = rating.split('/')[0].split('out')[0].strip()
+    if rating[-1] == '.':
+        rating = rating[:-1]
+    return rating
+
+
+def extract_rating(rating):
+    try:
+        assist1 = float(
+            clean_continuation(rating.split("Assistant 1:")[1].split("\n")[0]))
+        assist2 = float(
+            clean_continuation(rating.split("Assistant 2:")[1].split("\n")[0]))
+        return assist1-assist2
+    except ValueError as err:
+        print(err)
+        return None
+    except IndexError as err:
+        print(err)
+        return None
+
+
+def get_eval(system_prompt, user_prompt: str):
+    ratings = [-1 for _ in range(3)]
+    for rating_idx in range(3):
+        for _ in range(MAX_API_RETRY):
+            try:
+                messages = [{"role": "system", "content": system_prompt}] if system_prompt else []
+                messages.append({"role": "user", "content": user_prompt})
+                response = openai.ChatCompletion.create(
+                    model="gpt-3.5-turbo",
+                    messages=messages,
+                    temperature=0.0,
+                    max_tokens=512,
+                )
+
+                ratings[rating_idx] = response["choices"][0]["message"]["content"]
+
+            except Exception as e:
+                print(e)
+                sleep(REQ_TIME_GAP)
+
+        raise Exception(f"Failed after {MAX_API_RETRY} retries.")
```

Comment:
This exception would get triggered every time because unlike in https://github.com/CarperAI/autocrit/blob/7f89f3e583e232155e72cc576d54b55fcb8160c8/Rating/likert.py#L23-L29 where the result is returned immediately, multiple calls are needed to the endpoint here. Maybe it's better to initialize ratings as `None` and check for them down the line if needed.